site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,190 @@
+but it is always preferred to write partition position deletes because the writer already knows the exact partition to use after the scan.
+
+When performing a scan is too expensive or time-consuming, the compute engine can use equality deletes for faster write.
+It can convert an input predicate to global equality deletes, or convert it to equality predicates for each affected partition and write partition equality deletes.
+However, such conversion might not always be possible without scanning data (e.g. delete all data with `price > 2.33`). In those cases, using position deletes is preferred.
+
+For example, in a CDC pipeline, both partition position deletes and partition equality deletes are used.
+Consider an unpartitioned table with schema `(id bigint, category string, data string)` that has the following files with sequence numbers:
+
+```
+seq=0 file A: (1, 'c1', 'data1'), (2, 'c1', 'data2')
+seq=0 file B: (3, 'c2', 'data1'), (4, 'c2', 'data2')
+```
+
+The CDC pipeline writing to the table currently contains unflushed data `(1, 'c10', 'data10')` that will be committed as file C in the table.
+For a new delete predicate `id = 1`, the writer first checks the unflushed data index in memory and performs a position delete of file C at row position 0.
+It then writes an equality delete row `(1, NULL, NULL)` that is applied to all the existing data files A and B in the table.
+After the next commit checkpoint, the new table contains the following files:
+
+```
+seq=0 file A: (1, 'c1', 'data1'), (2, 'c1', 'data2')
+seq=0 file B: (3, 'c2', 'data1'), (4, 'c2', 'data2')
+seq=1 file C: (1, 'c10', 'data10')
+seq=1 position delete D: ('C', 0)
+seq=1 equality delete E: (1, NULL, NULL)
+```
+
+### Reading data with delete files
+
+During Iceberg's scan file planning phase, a delete file index is constructed to filter the delete files associated with each data file using the following rules:
+
+1. equality delete files are applied to data files of strictly lower sequence numbers
+2. position delete files are applied to data files of equal or lower sequence numbers
+3. further pruning is performed by comparing the partition and column statistics information of each pair of delete and data file. Therefore, for a partitioned table, partition deletes are always preferred to global deletes.
+
+In the CDC example in the last section, position delete D is applied to file C, and equality delete E is applied to file A.
+
+After the planning phase, each data file to read is associated with a set of delete files to merge with.
+In general, position deletes are easier to merge, because they are already sorted by file path and row position when writing.
+Applying position deletes to a data file can be viewed as merging two sorted lists, which can be done efficiently.
+In contrast, applying equality deletes to a data files requires loading all rows to memory and checking every row in a data file against every equality predicate.
```

Comment:
By `equality predicate` do you mean `equality delete predicate` ?

Reply:
```suggestion
In contrast, applying equality deletes to a data files requires loading all rows to memory and checking every row in a data file against every equality delete predicate.
```